tests/security/apparmor/aa_easyprof.pm

```diff
@@ -0,0 +1,84 @@
+# Copyright (C) 2018 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+# Summary: AppArmor profile generation made easy testing
+# Maintainer: Wes <whdu@suse.com>
+# Tags: poo#36895, tc#1621144
+
+use strict;
+use base "apparmortest";
+use testapi;
+use utils;
+use version_utils qw(is_sle is_leap);
+
+sub run {
+    my ($self) = @_;
+
+    my $aa_easyprof_jsonopt;
+    my $aa_easyprof_out;
+    my $aa_easyprof_tmp_json = "/tmp/manifest.json";
+    my $aa_easyprof_result   = "/tmp/output";
+    my @aa_easyprof_cmd;
+    my $aa_easyprof_cmd_convert;
+
+    # For new apparmor version, easyprof support convert from a manifest
+    # json file
+    if (is_sle('<15') or is_leap('<15.0')) {    # apparmor < 2.11.95
+        $aa_easyprof_out = $aa_easyprof_result;
+    }
+    else {                                      # apparmor >= 2.11.95
+        $aa_easyprof_out         = $aa_easyprof_tmp_json;
+        $aa_easyprof_jsonopt     = "--output-format=json";
+        $aa_easyprof_cmd_convert = "aa-easyprof --manifest=$aa_easyprof_out > $aa_easyprof_result";
+    }
+
+    @aa_easyprof_cmd = (
+        'aa-easyprof',                               "$aa_easyprof_jsonopt",
+        '--template=user-application',               '--policy-groups=opt-application,user-application',
+        '--abstractions="python,audio"',             '--read-path="/usr/share/foo/*"',
+        '--write-path="/opt/foo/tmp/"',              '--write-path="/opt/foo/log/"',
+        '--template-var="@{APPNAME}=foo"',           '--author="SUSE Tester"',
+        '--copyright="Copyright 2018, SUSE Tester"', '--comment="AppArmor is easy with aa-easyprof"',
+        "/usr/bin/foo > $aa_easyprof_out"
+    );
+
+    assert_script_run(join(' ', @aa_easyprof_cmd));
+    upload_logs($aa_easyprof_out);
+
+    if ($aa_easyprof_cmd_convert) {
+        assert_script_run($aa_easyprof_cmd_convert);
+        assert_script_run("rm -f $aa_easyprof_out");
```

Comment:
why you deleting log file ? and why you are doing this only for newer apparmor version ?

Reply:
For the older version, it only have one profile output fie, which will be cleared at the end of script.

Reply:
what older version ? you creating new test here

Reply:
The old apparmor version, there is no json file generated. But I will follow your suggestion to keep these output files there. This could make it simpler and faster.